be/src/olap/delete_bitmap_calculator.cpp

```diff
@@ -209,16 +211,18 @@ Status MergeIndexDeleteBitmapCalculator::calculate_one(RowLocation& loc) {
 }
 
 Status MergeIndexDeleteBitmapCalculator::calculate_all(DeleteBitmapPtr delete_bitmap) {
-    RowLocation loc;
-    while (true) {
-        auto st = calculate_one(loc);
-        if (st.is<ErrorCode::END_OF_FILE>()) {
-            break;
+    RETURN_IF_CATCH_EXCEPTION({
```

Comment:
you should consider to process the exception for method `MergeIndexDeleteBitmapCalculator::init` as well.
`_heap->push(&_contexts.back());` might throw exception there.